specification/search/data-plane/Azure.Search/preview/2025-05-01-preview/searchservice.json

```diff
@@ -0,0 +1,14715 @@
+              "example": "PT1H",
+              "default": "PT1H",
+              "description": "Specifies the duration for which the generated SAS URL will remain valid, formatted as an ISO 8601 duration."
+            }
+          }
+        },
+        "assetStore": {
+          "$ref": "#/definitions/SearchAssetStore",
+          "description": "The definition of an asset store that stores the assets."
+        }
+      }
+    },
+    "SearchAssetStoreType": {
+      "type": "string",
+      "enum": [
+        "azureblob"
+      ],
+      "x-ms-enum": {
+        "name": "SearchAssetStoreType",
+        "modelAsString": true,
+        "values": [
+          {
+            "value": "azureblob",
+            "name": "AzureBlob",
+            "description": "Indicates an Azure Blob asset store."
+          }
+        ]
+      },
+      "description": "Defines the type of an asset store."
+    },
+    "AssetStoreCredentials": {
+      "type": "object",
+      "properties": {
+        "connectionString": {
+          "externalDocs": {
+            "url": "https://to-be-created"
+          },
+          "type": "string",
+          "description": "The connection string for the asset store. Set to `<unchanged>` (with brackets) if you don't want the connection string updated. Set to `<redacted>` if you want to remove the connection string value from the assset store."
+        }
+      },
+      "description": "Represents credentials that can be used to connect to an asset store."
+    },
+    "SearchAssetStoreContainer": {
+      "type": "object",
+      "properties": {
+        "name": {
+          "type": "string",
+          "description": "The name of the container in the asset store that will be used to store assets."
+        }
+      },
+      "required": [
+        "name"
+      ],
+      "description": "The container in the asset store that will be used to store assets."
+    },
+    "SearchAssetStore": {
+      "type": "object",
+      "properties": {
+        "type": {
+          "$ref": "#/definitions/SearchAssetStoreType",
+          "description": "The type of the asset store."
+        },
+        "credentials": {
+          "$ref": "#/definitions/AssetStoreCredentials",
+          "description": "Credentials for the asset store."
+        },
+        "container": {
+          "$ref": "#/definitions/SearchAssetStoreContainer",
+          "description": "The data container for the asset store."
+        },
+        "identity": {
+          "$ref": "#/definitions/SearchIndexerDataIdentity",
+          "x-nullable": true,
+          "description": "An explicit managed identity to use for this asset store. If not specified and the connection string is a managed identity, the system-assigned managed identity is used. If not specified, the value remains unchanged. If \"none\" is specified, the value of this property is cleared."
+        }
+      },
+      "required": [
+        "name",
+        "type",
+        "credentials",
```

Comment:
Is credentials and identity an exclusive or (xor)? If they are, credentials shouldn't be required here as it basically puts consumers using identity in a spot that won't work

Reply:
Asset store was dropped